### DISCUSSION TITLE
Question - check someone subscribed to channel?

### DISCUSSION THREAD

**Discussion by rst630** [beyondcode/laravel-websockets#912] [Q&A]
I have thausands of event for broadcasting and huge load on server queue. Most channels has't subscribers, how I can check it before triggering event?

**Comment by LAGGOUNE-Walid** [comment]
Hello , this is not documented so you need to discover it the hard way . 
So lets start by the HTTP API provided by this package.
https://github.com/beyondcode/laravel-websockets/blob/master/src/Server/Router.php#L71
We are interested on this line of code
```php
$this->get('/apps/{appId}/channels/{channelName}', config('websockets.handlers.fetch_channel'));
```
The controller responsible for this route is *FetchChannel* :
https://github.com/beyondcode/laravel-websockets/blob/master/config/websockets.php#L278
https://github.com/beyondcode/laravel-websockets/blob/605a7fa71ddabfffe381b079acc21403c7547ea2/src/API/FetchChannel.php#L9

So you can make an HTTP GET to the url : 
_http://localhost:6002/apps/YOUR_APP_ID/channels/CHANNEL_NAME_
But this will give you an error , **Invalid auth signature provided**
The auth signature is generated like this : 
https://github.com/beyondcode/laravel-websockets/blob/master/src/API/Controller.php#L264

**Note: I only tested this using postman.**

This is my test : 
![alt text](https://i.ibb.co/mbCQcnN/Screenshot-from-2022-07-01-18-10-08.png)

i wish this will help you .
